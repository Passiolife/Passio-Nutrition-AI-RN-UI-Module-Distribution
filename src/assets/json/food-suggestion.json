[
  {
    "type": "breakfast",
    "passioID": "MEA0013"
  },
  {
    "type": "breakfast",
    "passioID": "ING0090"
  },
  {
    "type": "breakfast",
    "passioID": "ING0124"
  },
  {
    "type": "breakfast",
    "passioID": "ING0128"
  },
  {
    "type": "breakfast",
    "passioID": "VEG0235"
  },
  {
    "type": "breakfast",
    "passioID": "VEG0368"
  },
  {
    "type": "breakfast",
    "passioID": "VEG2075"
  },
  {
    "type": "breakfast",
    "passioID": "BAK0176"
  },
  {
    "type": "breakfast",
    "passioID": "VEG0038"
  },
  {
    "type": "breakfast",
    "passioID": "BAK0307"
  },
  {
    "type": "breakfast",
    "passioID": "BAK0222"
  },
  {
    "type": "breakfast",
    "passioID": "1000367"
  },
  {
    "type": "breakfast",
    "passioID": "GRA0027"
  },
  {
    "type": "lunch",
    "passioID": "ING0063"
  },
  {
    "type": "lunch",
    "passioID": "BEV0036"
  },
  {
    "type": "lunch",
    "passioID": "BEV0147"
  },
  {
    "type": "lunch",
    "passioID": "MEA0008"
  },
  {
    "type": "lunch",
    "passioID": "VEG0118"
  },
  {
    "type": "lunch",
    "passioID": "VEG0233"
  },
  {
    "type": "lunch",
    "passioID": "VEG0029"
  },
  {
    "type": "lunch",
    "passioID": "DAI0031"
  },
  {
    "type": "lunch",
    "passioID": "1001399"
  },
  {
    "type": "lunch",
    "passioID": "DAI0054"
  },
  {
    "type": "lunch",
    "passioID": "VEG0023"
  },
  {
    "type": "lunch",
    "passioID": "VEG0070"
  },
  {
    "type": "lunch",
    "passioID": "VEG0025"
  },
  {
    "type": "lunch",
    "passioID": "VEG3319"
  },
  {
    "type": "dinner",
    "passioID": "SEA0013"
  },
  {
    "type": "dinner",
    "passioID": "SEA0006"
  },
  {
    "type": "dinner",
    "passioID": "SEA0001"
  },
  {
    "type": "dinner",
    "passioID": "MID0123"
  },
  {
    "type": "dinner",
    "passioID": "MEA4424"
  },
  {
    "type": "dinner",
    "passioID": "DAI0080"
  },
  {
    "type": "dinner",
    "passioID": "DAI0137"
  },
  {
    "type": "dinner",
    "passioID": "DAI0110"
  },
  {
    "type": "dinner",
    "passioID": "VEG9922"
  },
  {
    "type": "dinner",
    "passioID": "VEG0122"
  },
  {
    "type": "dinner",
    "passioID": "VEG0494"
  },
  {
    "type": "dinner",
    "passioID": "VEG0241"
  },
  {
    "type": "dinner",
    "passioID": "VEG0104"
  },
  {
    "type": "snack",
    "passioID": "MEA0052"
  },
  {
    "type": "snack",
    "passioID": "MEA0135"
  },
  {
    "type": "snack",
    "passioID": "MEA9015"
  },
  {
    "type": "snack",
    "passioID": "MEA0221"
  },
  {
    "type": "snack",
    "passioID": "MEA0027"
  },
  {
    "type": "snack",
    "passioID": "MEA0205"
  },
  {
    "type": "snack",
    "passioID": "NUT0001"
  },
  {
    "type": "snack",
    "passioID": "NUT0005"
  },
  {
    "type": "snack",
    "passioID": "NUT0023"
  },
  {
    "type": "snack",
    "passioID": "NUT0036"
  },
  {
    "type": "snack",
    "passioID": "NUT0022"
  },
  {
    "type": "snack",
    "passioID": "VEG0052"
  },
  {
    "type": "snack",
    "passioID": "BAK0059"
  },
  {
    "type": "snack",
    "passioID": "VEG0399"
  },
  {
    "type": "snack",
    "passioID": "VEG0294"
  }
]