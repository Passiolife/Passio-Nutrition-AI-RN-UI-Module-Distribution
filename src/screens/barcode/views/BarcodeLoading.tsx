import React from 'react';
import { StyleSheet, View } from 'react-native';

import { Card, Text } from '../../../components';
import { screenHeight, screenWidth } from '../../../utils';
interface Props {}

export const BarcodeLoading = ({}: Props) => {
  return (
    <View style={styles.container}>
      <Card style={styles.card}>
        <View style={styles.contentContainer}>
          <Text weight="500" size="_14px" color="white" style={styles.title}>
            Scanning...
          </Text>
          <Text color="white" weight="400" style={styles.description}>
            Place your barcode within the frame
          </Text>
        </View>
      </Card>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: screenHeight - 250,
  },
  card: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 16,
    width: screenWidth - 16,
    paddingHorizontal: 12,
    marginHorizontal: 8,
    alignContent: 'center',
    backgroundColor: 'rgba(107, 114, 128, 0.75)',
    alignSelf: 'center',
  },
  contentContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    justifyContent: 'center',
    alignContent: 'center',
    alignItems: 'center',
  },
  icon: {
    height: 32,
    width: 32,
    marginVertical: 8,
  },
  title: {
    marginVertical: 3,
    textAlign: 'center',
  },
  description: {
    marginBottom: 16,
    marginVertical: 8,
    fontSize: 14,
    textAlign: 'center',
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignSelf: 'center',
    alignItems: 'center',
  },
  buttonContainers: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignContent: 'space-around',
    marginVertical: 16,
  },
  iconContainer: {
    alignItems: 'center',
    flex: 1,
  },
  button: {
    flex: 1,
    marginHorizontal: 4,
  },
});

export default BarcodeLoading;
