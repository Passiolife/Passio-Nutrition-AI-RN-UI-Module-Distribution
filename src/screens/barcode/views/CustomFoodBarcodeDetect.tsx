import React from 'react';
import { Image, StyleSheet, View } from 'react-native';

import { BasicButton, Card, Text } from '../../../components';
import { ICONS } from '../../../assets';
import { screenHeight } from '../../../utils';

interface Props {
  onCreateCustomWithoutBarcodePress?: () => void;
  onCancelPress?: () => void;
  onViewExistingPress?: () => void;
}

export const CustomFoodBarcodeDetect = ({
  onCreateCustomWithoutBarcodePress,
  onCancelPress,
  onViewExistingPress,
}: Props) => {
  return (
    <View style={styles.container}>
      <Card style={styles.card}>
        <View style={styles.contentContainer}>
          <Image source={ICONS.barcode} style={styles.icon} />
          <Text weight="700" size="_20px" style={styles.title}>
            Custom Food Already Exists
          </Text>
          <Text color="gray500" weight="400" style={styles.description}>
            This barcode matches an existing item in your custom food list. You
            can use the existing item, or create a new food without the barcode
          </Text>
        </View>
        <View style={styles.buttonContainers}>
          <BasicButton
            onPress={onCancelPress}
            style={styles.button}
            small
            text={'Cancel'}
            secondary
          />
          <BasicButton
            onPress={onViewExistingPress}
            style={styles.button}
            small
            text={'View Existing Item'}
          />
        </View>
        <View style={styles.buttonContainer}>
          <BasicButton
            onPress={onCreateCustomWithoutBarcodePress}
            style={styles.button}
            small
            text={'Create Custom Food Without Barcode'}
          />
        </View>
      </Card>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: screenHeight / 3,
    alignItems: 'center',
    alignContent: 'center',
    alignSelf: 'center',
    justifyContent: 'center',
  },
  card: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 16,
    paddingHorizontal: 12,
    marginHorizontal: 8,
    alignContent: 'center',

    alignSelf: 'center',
  },
  contentContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    justifyContent: 'center',
    alignContent: 'center',
    alignItems: 'center',
  },
  icon: {
    height: 32,
    width: 32,
    marginVertical: 8,
  },
  title: {
    marginVertical: 3,
    textAlign: 'center',
  },
  description: {
    marginBottom: 16,
    marginVertical: 8,
    fontSize: 14,
    textAlign: 'center',
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignSelf: 'center',
    alignItems: 'center',
  },
  buttonContainers: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignContent: 'space-around',
    marginVertical: 16,
  },
  iconContainer: {
    alignItems: 'center',
    flex: 1,
  },
  button: {
    flex: 1,
    marginHorizontal: 4,
  },
});

export default CustomFoodBarcodeDetect;
