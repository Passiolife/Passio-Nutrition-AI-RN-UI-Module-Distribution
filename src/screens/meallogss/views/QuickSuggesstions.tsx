import React from 'react';
import {
  type StyleProp,
  StyleSheet,
  View,
  type ViewStyle,
  FlatList,
} from 'react-native';
import { QuickSuggestionItemView } from './QuickSuggestionItemView';
import { COLORS } from '../../../constants';
import type { QuickSuggestion } from '../../../models/QuickSuggestion';
import { Text } from '../../../components/texts/Text';

interface Props {
  style?: StyleProp<ViewStyle>;
  onFoodLogEditor: (quickSuggestion: QuickSuggestion) => void;
  onFoodLog: (quickSuggestion: QuickSuggestion) => void;
  quickSuggestedAttributes: Array<QuickSuggestion>;
}

export const QuickSuggestions = ({
  style,
  onFoodLogEditor,
  onFoodLog,
  quickSuggestedAttributes,
}: Props) => {
  const renderFooter = () => {
    return <View style={styles.footer} />;
  };

  return (
    <View style={[styles.itemsContainer, style]}>
      <Text
        weight="600"
        size="_18px"
        color="text"
        style={styles.quickSuggestionTextStyle}
      >
        Quick Suggestions
      </Text>
      <Text
        weight="500"
        size="_14px"
        color="secondaryText"
        style={styles.noQuickSuggestionTitle}
      >
        Drag tray up to view suggestions to log
      </Text>
      {quickSuggestedAttributes && (
        <FlatList
          style={styles.list}
          data={quickSuggestedAttributes}
          numColumns={2}
          ListFooterComponent={renderFooter}
          renderItem={({ item }: { item: QuickSuggestion }) => {
            return (
              <QuickSuggestionItemView
                foodName={item.foodName}
                imageName={item.iconID}
                onFoodLogEditor={() => onFoodLogEditor(item)}
                onFoodLog={() => onFoodLog(item)}
              />
            );
          }}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  itemsContainer: {
    backgroundColor: 'white',
  },
  footer: {
    height: 120,
  },
  list: {
    marginHorizontal: 16,
    marginVertical: 16,
  },
  quickSuggestionTextStyle: {
    alignSelf: 'center',
    marginTop: 4,
    paddingHorizontal: 16,
  },
  noQuickSuggestionTitle: {
    paddingHorizontal: 16,
    marginBottom: 16,
    alignSelf: 'center',
  },
  noQuickSuggestionDescriptions: {
    fontSize: 15,
    alignSelf: 'center',
    textAlign: 'justify',
    fontWeight: '400',
    paddingHorizontal: 32,
    color: COLORS.grey7,
  },
});
