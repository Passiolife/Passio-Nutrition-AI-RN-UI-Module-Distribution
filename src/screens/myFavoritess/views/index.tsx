export * from './FavoriteEmptyView';
export * from './FavoriteFoodLogView';
