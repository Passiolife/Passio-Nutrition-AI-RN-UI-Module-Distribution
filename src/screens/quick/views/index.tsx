export * from '../../../components/loader/ProgressLoadingView';
export * from './QuickScanningResultView';
export * from './QuickScanningActionView';
export * from './QuickScanningLoadingView';
