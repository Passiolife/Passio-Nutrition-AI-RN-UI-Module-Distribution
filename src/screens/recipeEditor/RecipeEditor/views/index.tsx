export * from './RecipeMacroView';
export * from './IngredientsView';
