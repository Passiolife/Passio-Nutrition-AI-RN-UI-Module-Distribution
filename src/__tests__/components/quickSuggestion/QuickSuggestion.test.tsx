// @ts-nocheck
import React from 'react';
import {
  BrandingProvider,
  ServicesProvider,
} from '@passiolife/nutrition-ai-ui-ux';
import { cleanup, render } from '@testing-library/react-native';
import { mockBranding, mockServices } from '../../provider/MockProviders';
import { QuickSuggestions } from '../../../screens/meallogss/views/QuickSuggesstions';
import type { QuickSuggestion } from '../../../models/QuickSuggestion';
import type { PassioIDAttributes } from '@passiolife/nutritionai-react-native-sdk-v3/src/sdk/v2';

type Mock = jest.Mock;

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useIsFocused: () => jest.fn(),
  };
});

jest.mock('@passiolife/nutritionai-react-native-sdk-v3', () => {
  const mockSDK = {
    getAttributesForPassioID(): Promise<PassioIDAttributes | null> {
      return require('../../assets/json/mock_passio_sdk_attribute.json');
    },
  };
  const mockPassioIcon = require('../../utils/MockPassioIcon').default;

  return {
    PassioSDK: mockSDK,
    PassioIconView: mockPassioIcon,
    IconSize: 'PX90',
    PassioIDEntityType: {
      group: 'group',
    },
  };
});

describe('QuickSuggestion', () => {
  afterEach(cleanup);

  it('should render correctly', async () => {
    const onFoodItemClickCall = jest.fn();
    const quickSuggestionComponent = renderComponent(
      require('../../utils/json/quickSuggestion.json'),
      onFoodItemClickCall
    );
    expect(quickSuggestionComponent.toJSON()).toMatchSnapshot();
  });

  it('should show no quick suggestion message ', async () => {
    const onFoodItemClickCall = jest.fn();
    const quickSuggestionComponent = renderComponent([], onFoodItemClickCall);
    expect(quickSuggestionComponent.toJSON()).toMatchSnapshot();
  });

  const renderComponent = (
    quickSuggestedAttributes: QuickSuggestion[],
    onFoodItemClickCall: Mock
  ) => {
    const quickSuggestions = render(
      <ServicesProvider services={mockServices}>
        <BrandingProvider branding={mockBranding}>
          <QuickSuggestions
            quickSuggestedAttributes={quickSuggestedAttributes}
            onFoodItemClickCall={onFoodItemClickCall}
          />
        </BrandingProvider>
      </ServicesProvider>
    );
    return quickSuggestions;
  };
});
