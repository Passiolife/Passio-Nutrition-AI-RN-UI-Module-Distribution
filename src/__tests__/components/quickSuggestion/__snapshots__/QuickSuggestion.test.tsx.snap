// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickSuggestion should render correctly 1`] = `
<View
  style={
    [
      {
        "height": 300,
        "paddingBottom": 80,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#4E4B66",
        "fontSize": 17,
        "fontWeight": "700",
        "paddingBottom": 16,
        "paddingHorizontal": 16,
      }
    }
  >
    Quick Suggestions
  </Text>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "height": 60,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignContent": "center",
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 24,
                  "flexDirection": "row",
                  "height": 45,
                  "margin": 8,
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "borderRadius": 16,
                    "marginLeft": 4,
                    "overflow": "hidden",
                    "width": 32,
                  }
                }
              >
                <Text>
                  mockPassioIcon
                </Text>
              </View>
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "alignSelf": "center",
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "lineHeight": 16,
                    "marginHorizontal": 5,
                    "marginRight": 10,
                    "maxWidth": 150,
                    "textTransform": "capitalize",
                  }
                }
              >
                banana
              </Text>
              <Image
                source={
                  {
                    "testUri": "../../../src/assets/icons/add_blue.png",
                  }
                }
                style={
                  {
                    "alignSelf": "center",
                    "height": 32,
                    "margin": 16,
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 60,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignContent": "center",
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 24,
                  "flexDirection": "row",
                  "height": 45,
                  "margin": 8,
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "borderRadius": 16,
                    "marginLeft": 4,
                    "overflow": "hidden",
                    "width": 32,
                  }
                }
              >
                <Text>
                  mockPassioIcon
                </Text>
              </View>
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "alignSelf": "center",
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "lineHeight": 16,
                    "marginHorizontal": 5,
                    "marginRight": 10,
                    "maxWidth": 150,
                    "textTransform": "capitalize",
                  }
                }
              >
                turmeric
              </Text>
              <Image
                source={
                  {
                    "testUri": "../../../src/assets/icons/add_blue.png",
                  }
                }
                style={
                  {
                    "alignSelf": "center",
                    "height": 32,
                    "margin": 16,
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`QuickSuggestion should show no quick suggestion message  1`] = `
<View
  style={
    [
      {
        "height": 300,
        "paddingBottom": 80,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#4E4B66",
        "fontSize": 17,
        "fontWeight": "700",
        "paddingBottom": 16,
        "paddingHorizontal": 16,
      }
    }
  >
    Quick Suggestions
  </Text>
  <View />
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#333333",
        "fontSize": 17,
        "fontWeight": "700",
        "marginTop": 50,
        "paddingHorizontal": 32,
      }
    }
  >
    You have no quick suggestions
  </Text>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#333333",
        "fontSize": 15,
        "fontWeight": "400",
        "paddingHorizontal": 32,
        "textAlign": "justify",
      }
    }
  >
    Recent items you log with quick scan,
 multiscan, or search will appear here.
  </Text>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "width": "100%",
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
